frontend/src/components/WofButton.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <button :class="`button button--${type}`" @click="onClick()">
+    <div v-if="this.$slots.icon" :class="this.$slots.default ? 'button--icon' : ''"><slot name="icon"/></div>
+    <slot/>
+  </button>
+</template>
+
+<script>
+export default {
+  name: "WofButton",
+  props: {
+    type: {
+      type: String,
+      required: true,
+      default: "accent",
+      validator(value) {
+        return ["accent", "positive", "warning", "error", "disabled"].includes(value)
+      }
+    }
+  },
+  methods: {
+    onClick: function(event) {
+      this.$emit("click", event)
+    }
+  }
+}
+</script>
+
+<style lang="less">
+@import './common.less';
+
```

Comment:
We are using Less. Please, use nested classes. (.button and inside all these selectors for others variants etc)

Reply:
done